src/Domain/Encryption/Crypter.spec.ts

```diff
@@ -0,0 +1,62 @@
+import 'reflect-metadata'
+
+import { Crypter } from './Crypter'
+
+describe('Crypter', () => {
+  const encryptionIterations = 1000
+  const userServerKey = 'my-private-key'
+
+  const createCrypter = () => new Crypter(encryptionIterations)
+
+  it('should encrypt and decrypt data', async () => {
```

Comment:
You want to also create a test against a universal AES-GCM value. That is, essentially, find a popular library that incorporates AES-GCM, look at their unit tests, and find what they are testing. Then create a test with that same IV/key, and data, and ensure it encrypts and decrypts to the same value. This is the only you can be certain your implementation is correct. Otherwise your implementation could be correctly decrypting an incorrectly encrypted value.